r/configure

```diff
@@ -91,6 +96,14 @@ else
       fi
       PKG_CFLAGS="-I$BREWDIR/opt/$PKG_BREW_NAME/include"
       PKG_LIBS="-L$BREWDIR/opt/$PKG_BREW_NAME/lib $PKG_LIBS"
+    else
+      VERSION=$(grep ^Version DESCRIPTION | sed s/Version:\ //)
+      ${R_HOME}/bin/Rscript tools/linuxlibs.R $VERSION $LOCAL_LIBARROW
+      PKG_CFLAGS="-I$(pwd)/libarrow/arrow-${VERSION}/include $PKG_CFLAGS"
```

Comment:
We could; I was starting from https://github.com/apache/arrow/blob/master/r/tools/winlibs.R which does this business on Windows (and was originally provided by Jeroen, who maintains https://github.com/rwinlib). Aside from cargo cult, I have found the directory versioning helps if you run this in your local git checkout, helps make sure that the C++ library version you use matches the R package version. For the installed package itself, it doesn't matter what the directory is called because this is all temporary.